templates/guides/client_secret_rotation.md

```diff
@@ -0,0 +1,134 @@
+---
+page_title: Zero downtime client credentials rotation
+description: |-
+  Achieve zero downtime client credentials rotation with the Auth0 Terraform provider.
+---
+
+# Achieving zero downtime client credentials rotation
+
+In this guide we'll show how to rotate a client's credentials to eliminate downtime for the impacted system.
+
+## Pre-requisites:
+
+- The system relies on the configuration to maintain either a list of client secrets or at least two separate
+configuration entries (CURRENT and NEXT) representing client secrets.
+- By default, the system uses the first client secret in the list or the CURRENT configuration entry. If an error is
+received during an exchange that requires client authentication and the error is known to be associated with a problem 
+related to client credentials, the system should be capable of retrying the operation using the next secret in the
+list or the NEXT configuration entry.
+- If an event occurs that forces the system to attempt to use the next secret in the list or the NEXT configuration
+entry and the operation succeeds when using the new secret, the system should discard the old secret and update the
+configuration in a way that the next secret (the one that succeeded) is considered to be the default/current one going
+forward.
+
+## Process (to rotate the credentials):
+
+### Rotating a client secret
+
+1. Generate a new value for the client secret on behalf of the system associated with the client application record.
+This value should have similar entropy to the client secret values generated by the Auth0 service, minimum 48 chars and
+valid chars are numbers, letters and `_`, `-`, `+`, `=`, `.` symbols.
```

Comment:
Is there a command or standard process for generating this that we can recommend people?

Reply:
The random provider could be used https://registry.terraform.io/providers/hashicorp/random/latest/docs.